Button: Replace remaining 40px default size violations

Part of #63871

Throughout the app, there are still 167 instances of `Button` that use the old 36px default size. For consistency, we need to switch all these to the new 40px size.

Since this is a manual process requiring visual inspection of all instances, I broke the task down into manageable chunks so multiple people can work on it in parallel.

## How to submit PRs

In the PR, post a review comment for each changed file including brief testing instructions and before/after screenshots.

See #64524 for an example.

## Chunked tasks by package

To pick up a task, write your name by the checklist item or post a comment to this issue.

- [x] block-directory (@vipul0425 #65467)

    <details><summary>Details</summary>
    <p>

    ```
    packages/block-directory/src/components/downloadable-block-list-item/index.js
      98:5  error  Button should have the `__next40pxDefaultSize` prop to opt-in to the new default size  no-restricted-syntax

    packages/block-directory/src/plugins/get-install-missing/index.js
      103:4  error  Button should have the `__next40pxDefaultSize` prop to opt-in to the new default size  no-restricted-syntax

    packages/block-directory/src/plugins/get-install-missing/install-button.js
      24:3  error  Button should have the `__next40pxDefaultSize` prop to opt-in to the new default size  no-restricted-syntax
    ```

    </p>
    </details> 

- [x] block-editor 1 (@vipul0425 #65034 )

    <details><summary>Details</summary>
    <p>

    ```
    packages/block-editor/src/components/block-breadcrumb/index.js
      68:6  error  Button should have the `__next40pxDefaultSize` prop to opt-in to the new default size  no-restricted-syntax
      98:6  error  Button should have the `__next40pxDefaultSize` prop to opt-in to the new default size  no-restricted-syntax

    packages/block-editor/src/components/block-compare/block-view.js
      33:5  error  Button should have the `__next40pxDefaultSize` prop to opt-in to the new default size  no-restricted-syntax

    packages/block-editor/src/components/block-edit/multiple-usage-warning.js
      26:5  error  Button should have the `__next40pxDefaultSize` prop to opt-in to the new default size  no-restricted-syntax
      35:5  error  Button should have the `__next40pxDefaultSize` prop to opt-in to the new default size  no-restricted-syntax

    packages/block-editor/src/components/block-list/block-invalid-warning.js
      103:6  error  Button should have the `__next40pxDefaultSize` prop to opt-in to the new default size  no-restricted-syntax

    packages/block-editor/src/components/block-mover/button.js
      131:5  error  Button should have the `__next40pxDefaultSize` prop to opt-in to the new default size  no-restricted-syntax

    packages/block-editor/src/components/block-mover/index.js
      89:7  error  Button should have the `__next40pxDefaultSize` prop to opt-in to the new default size  no-restricted-syntax
    ```

    </p>
    </details> 

- [x] block-editor 2 (@dhruvang21 #65308)

    <details><summary>Details</summary>
    <p>

    ```
    packages/block-editor/src/components/block-navigation/dropdown.js
      29:3  error  Button should have the `__next40pxDefaultSize` prop to opt-in to the new default size  no-restricted-syntax

    packages/block-editor/src/components/block-pattern-setup/setup-toolbar.js
      20:3  error  Button should have the `__next40pxDefaultSize` prop to opt-in to the new default size  no-restricted-syntax
      38:3  error  Button should have the `__next40pxDefaultSize` prop to opt-in to the new default size  no-restricted-syntax
      47:3  error  Button should have the `__next40pxDefaultSize` prop to opt-in to the new default size  no-restricted-syntax
      71:4  error  Button should have the `__next40pxDefaultSize` prop to opt-in to the new default size  no-restricted-syntax
      79:4  error  Button should have the `__next40pxDefaultSize` prop to opt-in to the new default size  no-restricted-syntax

    packages/block-editor/src/components/block-patterns-paging/index.js
      40:7  error  Button should have the `__next40pxDefaultSize` prop to opt-in to the new default size  no-restricted-syntax
      51:7  error  Button should have the `__next40pxDefaultSize` prop to opt-in to the new default size  no-restricted-syntax
      76:7  error  Button should have the `__next40pxDefaultSize` prop to opt-in to the new default size  no-restricted-syntax
      87:7  error  Button should have the `__next40pxDefaultSize` prop to opt-in to the new default size  no-restricted-syntax

    packages/block-editor/src/components/block-quick-navigation/index.js
      61:3  error  Button should have the `__next40pxDefaultSize` prop to opt-in to the new default size  no-restricted-syntax

    packages/block-editor/src/components/block-tools/block-selection-button.js
      259:9  error  Button should have the `__next40pxDefaultSize` prop to opt-in to the new default size  no-restricted-syntax
      276:7  error  Button should have the `__next40pxDefaultSize` prop to opt-in to the new default size  no-restricted-syntax
    ```

    </p>
    </details> 

- [x] block-editor 3 (@vipul0425 #65225)

    <details><summary>Details</summary>
    <p>

    ```
    packages/block-editor/src/components/block-variation-picker/index.js
      66:6  error  Button should have the `__next40pxDefaultSize` prop to opt-in to the new default size  no-restricted-syntax

    packages/block-editor/src/components/block-variation-transforms/index.js
      37:5  error  Button should have the `__next40pxDefaultSize` prop to opt-in to the new default size  no-restricted-syntax

    packages/block-editor/src/components/button-block-appender/index.js
      62:6  error  Button should have the `__next40pxDefaultSize` prop to opt-in to the new default size  no-restricted-syntax

    packages/block-editor/src/components/colors-gradients/dropdown.js
      91:4  error  Button should have the `__next40pxDefaultSize` prop to opt-in to the new default size  no-restricted-syntax

    packages/block-editor/src/components/global-styles/color-panel.js
      242:7  error  Button should have the `__next40pxDefaultSize` prop to opt-in to the new default size  no-restricted-syntax

    packages/block-editor/src/components/global-styles/filters-panel.js
      192:10  error  Button should have the `__next40pxDefaultSize` prop to opt-in to the new default size  no-restricted-syntax

    packages/block-editor/src/components/global-styles/shadow-panel-components.js
       54:6  error  Button should have the `__next40pxDefaultSize` prop to opt-in to the new default size  no-restricted-syntax
      103:5  error  Button should have the `__next40pxDefaultSize` prop to opt-in to the new default size  no-restricted-syntax
      158:4  error  Button should have the `__next40pxDefaultSize` prop to opt-in to the new default size  no-restricted-syntax
    ```

    </p>
    </details> 

- [x] block-editor 4 (@vipul0425 #65257)

    <details><summary>Details</summary>
    <p>

    ```
    packages/block-editor/src/components/inserter-listbox/item.js
      46:6  error  Button should have the `__next40pxDefaultSize` prop to opt-in to the new default size  no-restricted-syntax

    packages/block-editor/src/components/inserter/block-patterns-explorer/pattern-explorer-sidebar.js
      17:6  error  Button should have the `__next40pxDefaultSize` prop to opt-in to the new default size  no-restricted-syntax

    packages/block-editor/src/components/inserter/block-patterns-tab/index.js
      63:6  error  Button should have the `__next40pxDefaultSize` prop to opt-in to the new default size  no-restricted-syntax

    packages/block-editor/src/components/inserter/media-tab/media-preview.js
      105:6  error  Button should have the `__next40pxDefaultSize` prop to opt-in to the new default size  no-restricted-syntax
      115:6  error  Button should have the `__next40pxDefaultSize` prop to opt-in to the new default size  no-restricted-syntax

    packages/block-editor/src/components/inserter/media-tab/media-tab.js
      73:9  error  Button should have the `__next40pxDefaultSize` prop to opt-in to the new default size  no-restricted-syntax

    packages/block-editor/src/components/inserter/quick-inserter.js
      143:5  error  Button should have the `__next40pxDefaultSize` prop to opt-in to the new default size  no-restricted-syntax
    ```

    </p>
    </details> 

- [x] block-editor 5 (@PARTHVATALIYA #65361)

    <details><summary>Details</summary>
    <p>

    ```
    packages/block-editor/src/components/inspector-controls-tabs/index.js
      51:9  error  Button should have the `__next40pxDefaultSize` prop to opt-in to the new default size  no-restricted-syntax

    packages/block-editor/src/components/inspector-popover-header/index.js
      33:6  error  Button should have the `__next40pxDefaultSize` prop to opt-in to the new default size  no-restricted-syntax
      47:6  error  Button should have the `__next40pxDefaultSize` prop to opt-in to the new default size  no-restricted-syntax

    packages/block-editor/src/components/link-control/index.js
      410:9  error  Button should have the `__next40pxDefaultSize` prop to opt-in to the new default size  no-restricted-syntax
      472:6  error  Button should have the `__next40pxDefaultSize` prop to opt-in to the new default size  no-restricted-syntax
      480:6  error  Button should have the `__next40pxDefaultSize` prop to opt-in to the new default size  no-restricted-syntax

    packages/block-editor/src/components/link-control/settings-drawer.js
      27:4  error  Button should have the `__next40pxDefaultSize` prop to opt-in to the new default size  no-restricted-syntax

    packages/block-editor/src/components/list-view/block-select-button.js
      85:3  error  Button should have the `__next40pxDefaultSize` prop to opt-in to the new default size  no-restricted-syntax

    packages/block-editor/src/components/media-placeholder/index.js
       53:4   error  Button should have the `__next40pxDefaultSize` prop to opt-in to the new default size  no-restricted-syntax
       89:4   error  Button should have the `__next40pxDefaultSize` prop to opt-in to the new default size  no-restricted-syntax
      388:5   error  Button should have the `__next40pxDefaultSize` prop to opt-in to the new default size  no-restricted-syntax
      418:6   error  Button should have the `__next40pxDefaultSize` prop to opt-in to the new default size  no-restricted-syntax
      435:5   error  Button should have the `__next40pxDefaultSize` prop to opt-in to the new default size  no-restricted-syntax
      476:10  error  Button should have the `__next40pxDefaultSize` prop to opt-in to the new default size  no-restricted-syntax
      507:8   error  Button should have the `__next40pxDefaultSize` prop to opt-in to the new default size  no-restricted-syntax
    ```

    </p>
    </details> 

- [x] block-editor 6 (@mirka #65742)

    <details><summary>Details</summary>
    <p>

    ```
    packages/block-editor/src/components/skip-to-selected-block/index.js
      30:3  error  Button should have the `__next40pxDefaultSize` prop to opt-in to the new default size  no-restricted-syntax

    packages/block-editor/src/components/tool-selector/index.js
      43:5  error  Button should have the `__next40pxDefaultSize` prop to opt-in to the new default size  no-restricted-syntax

    packages/block-editor/src/components/url-input/button.js
      40:5  error  Button should have the `__next40pxDefaultSize` prop to opt-in to the new default size  no-restricted-syntax
      55:8  error  Button should have the `__next40pxDefaultSize` prop to opt-in to the new default size  no-restricted-syntax
      67:8  error  Button should have the `__next40pxDefaultSize` prop to opt-in to the new default size  no-restricted-syntax

    packages/block-editor/src/components/url-input/index.js
      541:7  error  Button should have the `__next40pxDefaultSize` prop to opt-in to the new default size  no-restricted-syntax

    packages/block-editor/src/hooks/layout.js
      322:6  error  Button should have the `__next40pxDefaultSize` prop to opt-in to the new default size  no-restricted-syntax

    packages/block-editor/src/layouts/flex.js
      248:7  error  Button should have the `__next40pxDefaultSize` prop to opt-in to the new default size  no-restricted-syntax
    ```

    </p>
    </details> 

- [x] block-library 1 (@hbhalodia https://github.com/WordPress/gutenberg/pull/65033)

    <details><summary>Details</summary>
    <p>

    ```
    packages/block-library/src/comments/edit/comments-legacy.js
      31:3  error  Button should have the `__next40pxDefaultSize` prop to opt-in to the new default size  no-restricted-syntax

    packages/block-library/src/embed/embed-placeholder.js
      68:7  error  Button should have the `__next40pxDefaultSize` prop to opt-in to the new default size  no-restricted-syntax
      76:7  error  Button should have the `__next40pxDefaultSize` prop to opt-in to the new default size  no-restricted-syntax

    packages/block-library/src/freeform/modal.js
       27:3  error  Button should have the `__next40pxDefaultSize` prop to opt-in to the new default size  no-restricted-syntax
      125:8  error  Button should have the `__next40pxDefaultSize` prop to opt-in to the new default size  no-restricted-syntax
      135:8  error  Button should have the `__next40pxDefaultSize` prop to opt-in to the new default size  no-restricted-syntax

    packages/block-library/src/missing/edit.js
      52:3  error  Button should have the `__next40pxDefaultSize` prop to opt-in to the new default size  no-restricted-syntax
    ```

    </p>
    </details> 

- [x] block-library 2 (@hbhalodia https://github.com/WordPress/gutenberg/pull/65075)

    <details><summary>Details</summary>
    <p>

    ```
    packages/block-library/src/navigation-link/link-ui.js
      312:4  error  Button should have the `__next40pxDefaultSize` prop to opt-in to the new default size  no-restricted-syntax

    packages/block-library/src/navigation/edit/deleted-navigation-warning.js
      18:7  error  Button should have the `__next40pxDefaultSize` prop to opt-in to the new default size  no-restricted-syntax

    packages/block-library/src/navigation/edit/index.js
      593:9  error  Button should have the `__next40pxDefaultSize` prop to opt-in to the new default size  no-restricted-syntax

    packages/block-library/src/navigation/edit/navigation-menu-delete-control.js
      21:4  error  Button should have the `__next40pxDefaultSize` prop to opt-in to the new default size  no-restricted-syntax

    packages/block-library/src/navigation/edit/placeholder/index.js
      78:8  error  Button should have the `__next40pxDefaultSize` prop to opt-in to the new default size  no-restricted-syntax

    packages/block-library/src/navigation/edit/responsive-wrapper.js
       81:5  error  Button should have the `__next40pxDefaultSize` prop to opt-in to the new default size  no-restricted-syntax
      104:7  error  Button should have the `__next40pxDefaultSize` prop to opt-in to the new default size  no-restricted-syntax
    ```

    </p>
    </details> 

- [x] block-library 3 (@hbhalodia , https://github.com/WordPress/gutenberg/pull/65110)

    <details><summary>Details</summary>
    <p>

    ```
    packages/block-library/src/page-list/convert-to-links-modal.js
      34:5  error  Button should have the `__next40pxDefaultSize` prop to opt-in to the new default size  no-restricted-syntax
      42:5  error  Button should have the `__next40pxDefaultSize` prop to opt-in to the new default size  no-restricted-syntax

    packages/block-library/src/page-list/edit.js
      344:7  error  Button should have the `__next40pxDefaultSize` prop to opt-in to the new default size  no-restricted-syntax

    packages/block-library/src/post-comments-form/form.js
      86:5  error  Button should have the `__next40pxDefaultSize` prop to opt-in to the new default size  no-restricted-syntax

    packages/block-library/src/post-featured-image/edit.js
      310:7  error  Button should have the `__next40pxDefaultSize` prop to opt-in to the new default size  no-restricted-syntax

    packages/block-library/src/site-logo/edit.js
      629:11  error  Button should have the `__next40pxDefaultSize` prop to opt-in to the new default size  no-restricted-syntax
      676:8   error  Button should have the `__next40pxDefaultSize` prop to opt-in to the new default size  no-restricted-syntax

    packages/block-library/src/social-link/edit.js
      83:5  error  Button should have the `__next40pxDefaultSize` prop to opt-in to the new default size  no-restricted-syntax
    ```

    </p>
    </details> 

- [x] block-library 4 (@hbhalodia #65143)

    <details><summary>Details</summary>
    <p>

    ```
    packages/block-library/src/query/edit/enhanced-pagination-modal.js
      78:6  error  Button should have the `__next40pxDefaultSize` prop to opt-in to the new default size  no-restricted-syntax

    packages/block-library/src/query/edit/query-placeholder.js
      81:6  error  Button should have the `__next40pxDefaultSize` prop to opt-in to the new default size  no-restricted-syntax
      91:5  error  Button should have the `__next40pxDefaultSize` prop to opt-in to the new default size  no-restricted-syntax

    packages/block-library/src/template-part/edit/placeholder.js
      77:6  error  Button should have the `__next40pxDefaultSize` prop to opt-in to the new default size  no-restricted-syntax
      88:5  error  Button should have the `__next40pxDefaultSize` prop to opt-in to the new default size  no-restricted-syntax

    packages/block-library/src/video/edit.js
      234:10  error  Button should have the `__next40pxDefaultSize` prop to opt-in to the new default size  no-restricted-syntax
      264:9   error  Button should have the `__next40pxDefaultSize` prop to opt-in to the new default size  no-restricted-syntax

    packages/block-library/src/video/tracks-editor.js
       60:6  error  Button should have the `__next40pxDefaultSize` prop to opt-in to the new default size  no-restricted-syntax
      149:7  error  Button should have the `__next40pxDefaultSize` prop to opt-in to the new default size  no-restricted-syntax
      179:7  error  Button should have the `__next40pxDefaultSize` prop to opt-in to the new default size  no-restricted-syntax
    ```

    </p>
    </details> 

- [x] customize-widgets, edit-post (@ciampo https://github.com/WordPress/gutenberg/pull/65807)

    <details><summary>Details</summary>
    <p>

    ```
    packages/customize-widgets/src/components/error-boundary/index.js
      14:3  error  Button should have the `__next40pxDefaultSize` prop to opt-in to the new default size  no-restricted-syntax

    packages/customize-widgets/src/components/inserter/index.js
      39:5  error  Button should have the `__next40pxDefaultSize` prop to opt-in to the new default size  no-restricted-syntax

    packages/customize-widgets/src/components/welcome-guide/index.js
      45:4  error  Button should have the `__next40pxDefaultSize` prop to opt-in to the new default size  no-restricted-syntax

    packages/edit-post/src/components/back-button/fullscreen-mode-close.js
      93:4  error  Button should have the `__next40pxDefaultSize` prop to opt-in to the new default size  no-restricted-syntax

    packages/edit-post/src/components/init-pattern-modal/index.js
      90:9  error  Button should have the `__next40pxDefaultSize` prop to opt-in to the new default size  no-restricted-syntax

    packages/edit-post/src/components/preferences-modal/enable-custom-fields.js
      41:4  error  Button should have the `__next40pxDefaultSize` prop to opt-in to the new default size  no-restricted-syntax
    ```

    </p>
    </details> 

- [x] edit-site 1 (@hbhalodia #65226)

    <details><summary>Details</summary>
    <p>

    ```
    packages/edit-site/src/components/add-new-template/add-custom-template-modal-content.js
      45:5  error  Button should have the `__next40pxDefaultSize` prop to opt-in to the new default size  no-restricted-syntax

    packages/edit-site/src/components/add-new-template/index.js
      109:3  error  Button should have the `__next40pxDefaultSize` prop to opt-in to the new default size  no-restricted-syntax

    packages/edit-site/src/components/editor-canvas-container/index.js
      138:8  error  Button should have the `__next40pxDefaultSize` prop to opt-in to the new default size  no-restricted-syntax

    packages/edit-site/src/components/editor/index.js
      260:11  error  Button should have the `__next40pxDefaultSize` prop to opt-in to the new default size  no-restricted-syntax

    packages/edit-site/src/components/error-boundary/warning.js
      12:3  error  Button should have the `__next40pxDefaultSize` prop to opt-in to the new default size  no-restricted-syntax
    ```

    </p>
    </details> 

- [x] edit-site 2 (@hbhalodia #65258)

    <details><summary>Details</summary>
    <p>

    ```
    packages/edit-site/src/components/global-styles/font-library-modal/font-card.js
      30:3  error  Button should have the `__next40pxDefaultSize` prop to opt-in to the new default size  no-restricted-syntax

    packages/edit-site/src/components/global-styles/font-library-modal/font-collection.js
      474:8  error  Button should have the `__next40pxDefaultSize` prop to opt-in to the new default size  no-restricted-syntax

    packages/edit-site/src/components/global-styles/font-library-modal/installed-fonts.js
      439:8  error  Button should have the `__next40pxDefaultSize` prop to opt-in to the new default size  no-restricted-syntax
      449:7  error  Button should have the `__next40pxDefaultSize` prop to opt-in to the new default size  no-restricted-syntax

    packages/edit-site/src/components/global-styles/font-library-modal/upload-fonts.js
      232:8  error  Button should have the `__next40pxDefaultSize` prop to opt-in to the new default size  no-restricted-syntax

    packages/edit-site/src/components/global-styles/palette.js
      89:6  error  Button should have the `__next40pxDefaultSize` prop to opt-in to the new default size  no-restricted-syntax

    packages/edit-site/src/components/global-styles/screen-revisions/revisions-buttons.js
      164:7  error  Button should have the `__next40pxDefaultSize` prop to opt-in to the new default size  no-restricted-syntax

    packages/edit-site/src/components/global-styles/shadows-edit-panel.js
      236:9  error  Button should have the `__next40pxDefaultSize` prop to opt-in to the new default size  no-restricted-syntax
      248:9  error  Button should have the `__next40pxDefaultSize` prop to opt-in to the new default size  no-restricted-syntax
      383:8  error  Button should have the `__next40pxDefaultSize` prop to opt-in to the new default size  no-restricted-syntax
      396:9  error  Button should have the `__next40pxDefaultSize` prop to opt-in to the new default size  no-restricted-syntax
    ```

    </p>
    </details> 

- [x] edit-site 3 (@hbhalodia #65309)

    <details><summary>Details</summary>
    <p>

    ```
    packages/edit-site/src/components/page-patterns/fields.js
      135:6  error  Button should have the `__next40pxDefaultSize` prop to opt-in to the new default size  no-restricted-syntax

    packages/edit-site/src/components/save-panel/index.js
      152:5  error  Button should have the `__next40pxDefaultSize` prop to opt-in to the new default size  no-restricted-syntax

    packages/edit-site/src/components/sidebar-button/index.js
      13:3  error  Button should have the `__next40pxDefaultSize` prop to opt-in to the new default size  no-restricted-syntax

    packages/edit-site/src/components/sidebar-dataviews/add-new-view.js
       90:6  error  Button should have the `__next40pxDefaultSize` prop to opt-in to the new default size  no-restricted-syntax
      101:6  error  Button should have the `__next40pxDefaultSize` prop to opt-in to the new default size  no-restricted-syntax

    packages/edit-site/src/components/site-hub/index.js
       64:7  error  Button should have the `__next40pxDefaultSize` prop to opt-in to the new default size  no-restricted-syntax
       82:8  error  Button should have the `__next40pxDefaultSize` prop to opt-in to the new default size  no-restricted-syntax
      103:8  error  Button should have the `__next40pxDefaultSize` prop to opt-in to the new default size  no-restricted-syntax
      151:7  error  Button should have the `__next40pxDefaultSize` prop to opt-in to the new default size  no-restricted-syntax
      172:8  error  Button should have the `__next40pxDefaultSize` prop to opt-in to the new default size  no-restricted-syntax
      188:8  error  Button should have the `__next40pxDefaultSize` prop to opt-in to the new default size  no-restricted-syntax
    ```

    </p>
    </details> 

- [x] edit-widgets (@dhruvang21 #65367)

    <details><summary>Details</summary>
    <p>

    ```
    packages/edit-widgets/src/components/error-boundary/index.js
      14:3  error  Button should have the `__next40pxDefaultSize` prop to opt-in to the new default size  no-restricted-syntax

    packages/edit-widgets/src/components/secondary-sidebar/inserter-sidebar.js
      46:5  error  Button should have the `__next40pxDefaultSize` prop to opt-in to the new default size  no-restricted-syntax

    packages/edit-widgets/src/components/secondary-sidebar/list-view-sidebar.js
      53:5  error  Button should have the `__next40pxDefaultSize` prop to opt-in to the new default size  no-restricted-syntax

    packages/edit-widgets/src/components/sidebar/widget-areas.js
      68:7  error  Button should have the `__next40pxDefaultSize` prop to opt-in to the new default size  no-restricted-syntax
    ```

    </p>
    </details> 

- [x] editor 1 (@AKSHAT2802 #65068)

    <details><summary>Details</summary>
    <p>

    ```
    packages/editor/src/components/block-manager/index.js
      69:6  error  Button should have the `__next40pxDefaultSize` prop to opt-in to the new default size  no-restricted-syntax

    packages/editor/src/components/editor-history/redo.js
      27:3  error  Button should have the `__next40pxDefaultSize` prop to opt-in to the new default size  no-restricted-syntax

    packages/editor/src/components/editor-history/undo.js
      23:3  error  Button should have the `__next40pxDefaultSize` prop to opt-in to the new default size  no-restricted-syntax

    packages/editor/src/components/error-boundary/index.js
      32:3  error  Button should have the `__next40pxDefaultSize` prop to opt-in to the new default size  no-restricted-syntax

    packages/editor/src/components/post-excerpt/panel.js
      200:6  error  Button should have the `__next40pxDefaultSize` prop to opt-in to the new default size  no-restricted-syntax

    packages/editor/src/components/post-featured-image/index.js
      162:9   error  Button should have the `__next40pxDefaultSize` prop to opt-in to the new default size  no-restricted-syntax
      204:11  error  Button should have the `__next40pxDefaultSize` prop to opt-in to the new default size  no-restricted-syntax
      213:11  error  Button should have the `__next40pxDefaultSize` prop to opt-in to the new default size  no-restricted-syntax
    ```

    </p>
    </details> 

- [x] editor 2 (@AKSHAT2802 #65083)

    <details><summary>Details</summary>
    <p>

    ```
    packages/editor/src/components/post-format/index.js
      102:7  error  Button should have the `__next40pxDefaultSize` prop to opt-in to the new default size  no-restricted-syntax

    packages/editor/src/components/post-last-revision/index.js
      38:4  error  Button should have the `__next40pxDefaultSize` prop to opt-in to the new default size  no-restricted-syntax

    packages/editor/src/components/post-locked-modal/index.js
      259:8  error  Button should have the `__next40pxDefaultSize` prop to opt-in to the new default size  no-restricted-syntax
      268:7  error  Button should have the `__next40pxDefaultSize` prop to opt-in to the new default size  no-restricted-syntax

    packages/editor/src/components/post-publish-panel/index.js
      80:7  error  Button should have the `__next40pxDefaultSize` prop to opt-in to the new default size  no-restricted-syntax

    packages/editor/src/components/post-publish-panel/maybe-post-format-panel.js
      27:2  error  Button should have the `__next40pxDefaultSize` prop to opt-in to the new default size  no-restricted-syntax

    packages/editor/src/components/post-publish-panel/maybe-upload-media.js
      157:6  error  Button should have the `__next40pxDefaultSize` prop to opt-in to the new default size  no-restricted-syntax
    ```

    </p>
    </details> 

- [x] editor 3 (@AKSHAT2802 #65139)

    <details><summary>Details</summary>
    <p>

    ```
    packages/editor/src/components/post-taxonomies/hierarchical-term-selector.js
      430:6  error  Button should have the `__next40pxDefaultSize` prop to opt-in to the new default size  no-restricted-syntax

    packages/editor/src/components/post-taxonomies/most-used-terms.js
      64:7  error  Button should have the `__next40pxDefaultSize` prop to opt-in to the new default size  no-restricted-syntax

    packages/editor/src/components/post-template/classic-theme.js
      179:6  error  Button should have the `__next40pxDefaultSize` prop to opt-in to the new default size  no-restricted-syntax

    packages/editor/src/components/post-url/index.js
      112:9  error  Button should have the `__next40pxDefaultSize` prop to opt-in to the new default size  no-restricted-syntax

    packages/editor/src/components/save-publish-panels/index.js
      73:5  error  Button should have the `__next40pxDefaultSize` prop to opt-in to the new default size  no-restricted-syntax
      88:5  error  Button should have the `__next40pxDefaultSize` prop to opt-in to the new default size  no-restricted-syntax
    ```

    </p>
    </details> 

- [x] editor 4 (@AKSHAT2802 #65140)

    <details><summary>Details</summary>
    <p>

    ```
    packages/editor/src/components/start-template-options/index.js
      158:6  error  Button should have the `__next40pxDefaultSize` prop to opt-in to the new default size  no-restricted-syntax

    packages/editor/src/components/table-of-contents/index.js
      32:5  error  Button should have the `__next40pxDefaultSize` prop to opt-in to the new default size  no-restricted-syntax

    packages/editor/src/components/text-editor/index.js
      42:6  error  Button should have the `__next40pxDefaultSize` prop to opt-in to the new default size  no-restricted-syntax

    packages/editor/src/dataviews/actions/reset-post.tsx
      116:6  error  Button should have the `__next40pxDefaultSize` prop to opt-in to the new default size  no-restricted-syntax
      126:6  error  Button should have the `__next40pxDefaultSize` prop to opt-in to the new default size  no-restricted-syntax

    packages/editor/src/dataviews/actions/trash-post.tsx
      69:6  error  Button should have the `__next40pxDefaultSize` prop to opt-in to the new default size  no-restricted-syntax
      79:6  error  Button should have the `__next40pxDefaultSize` prop to opt-in to the new default size  no-restricted-syntax
    ```

    </p>
    </details> 

- [x] dataviews, list-reusable-blocks, nux, reusable-blocks (@mirka #65715)

    <details><summary>Details</summary>
    <p>

    ```
    packages/dataviews/src/dataforms-layouts/panel/index.tsx
      46:6  error  Button should have the `__next40pxDefaultSize` prop to opt-in to the new default size  no-restricted-syntax


    packages/list-reusable-blocks/src/components/import-dropdown/index.js
      19:5  error  Button should have the `__next40pxDefaultSize` prop to opt-in to the new default size  no-restricted-syntax

    packages/list-reusable-blocks/src/components/import-form/index.js
      86:4  error  Button should have the `__next40pxDefaultSize` prop to opt-in to the new default size  no-restricted-syntax


    packages/nux/src/components/dot-tip/index.js
      59:5  error  Button should have the `__next40pxDefaultSize` prop to opt-in to the new default size  no-restricted-syntax
      68:4  error  Button should have the `__next40pxDefaultSize` prop to opt-in to the new default size  no-restricted-syntax


    packages/reusable-blocks/src/components/reusable-blocks-menu-items/reusable-block-convert-button.js
      193:9  error  Button should have the `__next40pxDefaultSize` prop to opt-in to the new default size  no-restricted-syntax
      205:9  error  Button should have the `__next40pxDefaultSize` prop to opt-in to the new default size  no-restricted-syntax
    ```

    </p>
    </details>

--- a/.eslintrc.js
+++ b/.eslintrc.js
@@ -322,6 +322,7 @@ module.exports = {
 						'BorderBoxControl',
 						'BorderControl',
 						'BoxControl',
+						'Button',
 						'ComboboxControl',
 						'CustomSelectControl',
 						'DimensionControl',
@@ -352,14 +353,6 @@ module.exports = {
 						message:
 							'FormFileUpload should have the `__next40pxDefaultSize` prop to opt-in to the new default size.',
 					},
-					// Temporary rules until all existing components have the `__next40pxDefaultSize` prop.
-					...[ 'Button' ].map( ( componentName ) => ( {
-						// Not strict. Allows pre-existing __next40pxDefaultSize={ false } usage until they are all manually updated.
-						selector: `JSXOpeningElement[name.name="${ componentName }"]:not(:has(JSXAttribute[name.name="__next40pxDefaultSize"])):not(:has(JSXAttribute[name.name="size"]))`,
-						message:
-							componentName +
-							' should have the `__next40pxDefaultSize` prop to opt-in to the new default size.',
-					} ) ),
 				],
 			},
 		},

--- a/.eslintrc.js
+++ b/.eslintrc.js
@@ -344,7 +344,7 @@ module.exports = {
 						selector: `JSXOpeningElement[name.name="${ componentName }"]:not(:has(JSXAttribute[name.name="__next40pxDefaultSize"][value.expression.value!=false])):not(:has(JSXAttribute[name.name="size"][value.value!="default"]))`,
 						message:
 							componentName +
-							' should have the `__next40pxDefaultSize` prop to opt-in to the new default size.',
+							' should have the `__next40pxDefaultSize` prop when using the default size.',
 					} ) ),
 					{
 						// Falsy `__next40pxDefaultSize` without a `render` prop.

--- a/packages/editor/src/components/post-template/classic-theme.js
+++ b/packages/editor/src/components/post-template/classic-theme.js
@@ -177,8 +177,7 @@ function PostTemplateDropdownContent( { onClose } ) {
 			{ canEdit && onNavigateToEntityRecord && (
 				<p>
 					<Button
-						// TODO: Switch to `true` (40px size) if possible
-						__next40pxDefaultSize={ false }
+						__next40pxDefaultSize
 						variant="link"
 						onClick={ () => {
 							onNavigateToEntityRecord( {

